openapi_core/schema/schemas/models.py

```diff
@@ -577,19 +632,25 @@ def _validate_properties(self, value, one_of_schema=None,
             for prop_name in extra_props:
                 prop_value = value[prop_name]
                 self.additional_properties.validate(
-                    prop_value, custom_formatters=custom_formatters)
+                    prop_value,
+                    custom_formatters=custom_formatters,
+                    require_all_props=require_all_props
+                )
 
         for prop_name, prop in iteritems(all_props):
             try:
                 prop_value = value[prop_name]
             except KeyError:
-                if prop_name in all_req_props_names:
+                if (prop_name in all_req_props_names) or require_all_props:
```

Comment:
@reviewers This is the important line. All other changes are passing around the flag.